src/librustc/middle/expr_use_visitor.rs

```diff
@@ -578,14 +580,16 @@ impl<'a, 'gcx, 'tcx> ExprUseVisitor<'a, 'gcx, 'tcx> {
                     };
                 match overloaded_call_type {
                     FnMutOverloadedCall => {
+                        let r = self.tcx().mk_region(ty::ReScope(call_scope));
```

Comment:
I wonder if it's worth adding a helper for this, like `mk_scope_region(call_scope)`? Or something to make it short enough to not have to make short variable names like `r`.